Reviewer/src/main/java/com/softserveinc/reviewer/health/ReviewerHealthCheck.java

```diff
@@ -0,0 +1,95 @@
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class ReviewerHealthCheck extends HealthCheck {
+    private static final int CONNECTION_TIMEOUT = 250;
+    private static final int FUTURE_TIMEOUT = 300;
+    private static final String HEALTHY = "Service is healthy.";
+    private static final String UNHEALTHY = "Service is unhealthy.";
+    private static final String UNAVAILABLE = "Service unavailable at the moment.";
+    private static final String FAILED = "Failed to reach the Service.";
+    private final String switchBoardHealthCheckUrl;
+    private final String oracleHealthCheckUrl;
+    private final String elasticSearchHealthCheckUrl;
+    private final ExecutorService executorService = Executors.newFixedThreadPool(3);
+    private boolean healthy = true;
+
+    @Inject
+    public ReviewerHealthCheck(String switchBoardHealthCheckUrl, String oracleHealthCheckUrl, String elasticSearchHealthCheckUrl) {
+        this.switchBoardHealthCheckUrl = switchBoardHealthCheckUrl;
+        this.oracleHealthCheckUrl = oracleHealthCheckUrl;
+        this.elasticSearchHealthCheckUrl = elasticSearchHealthCheckUrl;
+    }
+
+    @Override
+    protected Result check() throws Exception {
+        Future<String> switchBoardFuture = executorService.submit(new CheckStatus(switchBoardHealthCheckUrl));
+        Future<String> oracleFuture = executorService.submit(new CheckStatus(oracleHealthCheckUrl));
+        Future<String> elasticSearchFuture = executorService.submit(new CheckStatus(elasticSearchHealthCheckUrl));
+        StringBuilder message = new StringBuilder();
+        try {
+            message.append(" switchboard: " + switchBoardFuture.get(FUTURE_TIMEOUT, TimeUnit.MILLISECONDS));
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            switchBoardFuture.cancel(true);
+            message.append(" switchboard: " + FAILED);
+            healthy = false;
+        }
+        try {
+            message.append(" oracle: " + oracleFuture.get(FUTURE_TIMEOUT, TimeUnit.MILLISECONDS));
+        } catch (InterruptedException | ExecutionException  | TimeoutException e) {
+            oracleFuture.cancel(true);
+            message.append(" oracle: " + FAILED);
+            healthy = false;
+        }
+        try {
+            message.append(" elasticsearch: " + elasticSearchFuture.get(FUTURE_TIMEOUT, TimeUnit.MILLISECONDS));
+        } catch (InterruptedException | ExecutionException  | TimeoutException e) {
+            elasticSearchFuture.cancel(true);
+            message.append(" elasticsearch: " + FAILED);
+            healthy = false;
+        }
+        return healthy ? Result.healthy(message.toString()) : Result.unhealthy(message.toString());
```

Comment:
* ```healthy``` shouldn't be a property of class. As you noted, it screws up results by affecting following requests.

Please don't use message for reporting combined statuses. When we need to check several underlying services, that should be done by registering different healthchecks. So that you could see status of each subsystem separately (in proper json node). Here we need to have a single top-level health check which **execution time is guaranteed**.

I wish you would accept varargs of tasks in the constructor instead of having hard-coded initialization in the class. So that we could provide any number of HealthCheck-related tasks into that class and let it execute with some given timeout.

As of now you execute tasks manually and sequentially. Please find the way to run checks all at once and have them running in parallel. Let them run for some limited time and then use combined result as a healthcheck status.